emily/cdk/lib/emily-stack-utils.ts

```diff
@@ -0,0 +1,151 @@
+        apiLambdas: [lambdaIdentifier: string, lambdaFunction: lambda.Function][],
+    ): ApiDefinition {
+
+        // This whole section is unfortunate but there's not a standard solution. The autogenerated OpenAPI
+        // template will always setup the `uri` to have injected substitution with `Fn::Sub`, and the only
+        // way to gather the ARN of the lambda as a string at build time is build the whole arn yourself which
+        // is worse than this solution; if the arn format changes the api will break without an obvious error.
+        // When we get the ARN from lambda.Function object it generates a template function that will gather
+        // the ARN and pretend it's a string, so you cannot substitute anything with it directly.
+        //
+        // In this loop we go through the OpenAPI json and inject it with a parameter such that the
+        // `Fn::Sub` can replace with our desired lambda's arn.
+        //
+        // https://repost.aws/knowledge-center/cloudformation-fn-sub-function
+        //
+        // If you look at the template that spawns from this you'll see that the `Fn::Sub` actually defines
+        // the key to be ANOTHER `Fn::XXX` template function that resolves to be the lambda's Arn.
+        //
+        // It would be nice if there were a solution that didn't involve doing this ourselves.
+        let apiJsonDefiniton = JSON.parse(fs.readFileSync(restApiPathOrFileDescriptor, 'utf-8'));
+        let paths = apiJsonDefiniton["paths"];
+        Object.keys(paths).forEach(path => {
+            let verbs = paths[path];
+            Object.keys(verbs).forEach(verb => {
+                if (Object.keys(verbs[verb]).includes("x-amazon-apigateway-integration")) {
+                    let awsIntegration = verbs[verb]["x-amazon-apigateway-integration"];
+                    let originalSubString: string = awsIntegration["uri"]["Fn::Sub"];
+                    apiLambdas.forEach(([lambdaIdentifier, lambdaFunction]) => {
+                        if (originalSubString.includes(`\${${lambdaIdentifier}}`)) {
+                            // If the identifier is present in the uri string then generate the template function to
+                            // replace the identifier with the lambda arn.
+                            //
+                            // This will incorrectly handle the already invalid case where two function ARNS are in the uri.
+                            // Handling multiple replacements is left as an exercize for the reader.
+                            apiJsonDefiniton["paths"][path][verb]["x-amazon-apigateway-integration"]["uri"]["Fn::Sub"] =
+                                [ originalSubString, { [lambdaIdentifier] : lambdaFunction.functionArn } ]
+                        }
+                    })
+                }
+            })
+        })
+
+        // Return the modified template as an ApiDefinition.
+        return ApiDefinition.fromInline(apiJsonDefiniton)
+    }
```

Comment:
I'm disappointed that this is the best option, and it relies on the openapi definition being well formatted, but we get some guarantees based on it being an OpenAPI spec. The errors that occur when the template is mangled are very obvious, which is what  bug here would result in.

Lets test everything... except not this please. If it crashes it's a design decision I swear